cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -195,6 +197,7 @@ protected void configure() {
           bind(MetricDatasetFactory.class).to(DefaultMetricDatasetFactory.class).in(Scopes.SINGLETON);
           bind(MetricStore.class).to(DefaultMetricStore.class);
           bind(DatasetFramework.class).to(InMemoryDatasetFramework.class).in(Scopes.SINGLETON);
+          bind(DatasetProvider.class).to(DefaultDatasetProvider.class);
```

Comment:
The dependency is UpgradeTool -> AppFabricServiceRuntimeModule -> AppLifecycleHttpHandler -> ApplicationLifecycleService.